src/client/sound/s_mix.c

```diff
@@ -90,12 +106,23 @@ static _Bool S_SpatializeChannel(s_channel_t *ch) {
 	const vec_t dist = VectorNormalize(delta) * attenuation;
 	const vec_t frac = dist / SOUND_MAX_DISTANCE;
 
-	ch->dist = (uint8_t) (Clamp(frac * 255.0, 0.0, 255.0));
+	ch->gain = 1.0 - frac;
 
-	const vec_t dot = DotProduct(r_view.right, delta);
-	const vec_t angle = acos(dot) * 180.0 / M_PI - 90.0;
+	// fade out frame sounds if they are no longer in the frame
+	if (ch->start_time) {
+		if (ch->play.flags & S_PLAY_FRAME) {
+			if (ch->frame != cl.frame.frame_num) {
+				const int32_t frame_diff = cl.frame.frame_num - ch->frame;
+				const uint32_t ms_diff = frame_diff * QUETOO_TICK_MILLIS;
 
```

Comment:
To match the rest of the code, I'd probably name these `frames` and `delta` respectively.